src/view/search/advance-filter-modal.tsx

```diff
@@ -0,0 +1,141 @@
+/* eslint-disable prettier/prettier */
+/*
+ * SPDX-FileCopyrightText: 2021 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import React, { FC, ReactElement, useState, useCallback, useMemo, useEffect } from 'react';
+import { CustomModal, Container, ChipProps } from '@zextras/carbonio-design-system';
+import { concat, filter, includes, map } from 'lodash';
+import { QueryChip, t } from '@zextras/carbonio-shell-ui';
+import moment from 'moment';
+import ModalFooter from '../../commons/modal-footer';
+import { ModalHeader } from '../../commons/modal-header';
+import KeywordRow from './parts/keyword-row';
+import FromDateToDateRow from './parts/from-date-to-date-row';
+
+type KeywordState = Array<{
+	id: string;
+	label: string;
+	hasAvatar?: boolean;
+	value?: string;
+	isQueryFilter?: boolean;
+	isGeneric?: boolean;
+	avatarIcon?: string;
+	avatarBackground?: ChipProps['background'];
+	hasError?: boolean;
+	error?: boolean;
+}>;
+
+type AdvancedFilterModalProps = {
+	open: boolean;
+	dateStart: number;
+	dateEnd: number;
+    setDateStart: (arg: number) => void;
+    setDateEnd: (arg: number) => void;
+	onClose: () => void;
+	query: Array<{
+		id: string;
+		label: string;
+		value?: string;
+		isGeneric?: boolean;
+		isQueryFilter?: boolean;
+	}>;
+	updateQuery: (arg: Array<QueryChip>) => void;
+};
+
+const AdvancedFilterModal: FC<AdvancedFilterModalProps> = ({
+	open,
+	onClose,
+	query,
+	updateQuery,
+	dateStart,
+	dateEnd,
+    setDateStart,
+    setDateEnd
+}): ReactElement => {
+	const [otherKeywords, setOtherKeywords] = useState<KeywordState>([]);
+
+	const [folder, setFolder] = useState<KeywordState>([]);
+	const [fromDate, setFromDate] = useState<any>(dateStart);
+	const [toDate, setToDate] = useState<any>(dateEnd);
+	const queryArray = useMemo(() => ['has:attachment', 'is:flagged', 'is:unread'], []);
+
+	useEffect(() => {
+		const updatedQuery = map(
+			filter(
+				query,
+				(v) =>
+					!includes(queryArray, v.label) &&
+					!/^Is:/.test(v.label) &&
+					!v.isQueryFilter
+			),
+			(q) => ({ ...q, hasAvatar: false })
+		);
+		setOtherKeywords(updatedQuery);
+	}, [query, queryArray]);
+    
+    const DEFAULT_DATE_START = moment().startOf('day').subtract(1, 'months').valueOf();
```

Comment:
move these two constants in the constants directory in a separated file so you can use it everywhere.

Reply:
moved, tnx!